### PR TITLE
Add Envoy Proxy configuration example

### LINKED ISSUE #1543: Cookbook example for Envoy deployment
With Nginx going out of fashion we should add an example for how to use [Envoy](https://www.envoyproxy.io) as a reverse-proxy with Mojolicious to the [cookbook](https://mojolicious.org/perldoc/Mojolicious/Guides/Cookbook#Nginx).

### PR DESCRIPTION
**Pull request by zakame**
This adds a simple reverse proxy example using Envoy.

Fixes #1543.

### CODE REVIEW

File: lib/Mojolicious/Guides/Cookbook.pod [hunk 1]
```diff
@@ -342,6 +342,51 @@ web features, such as WebSockets, are not available.
 
   ScriptAlias / /home/sri/my_app/script/my_app/
 
+=head2 Envoy Proxy
```
Comment (kraih) [inline lib/Mojolicious/Guides/Cookbook.pod:345]: Why does the header need a "Proxy" qualifier? Are there different ways to deploy Mojolicious apps with Envoy?

File: lib/Mojolicious/Guides/Cookbook.pod [hunk 2]
```diff
@@ -342,6 +342,51 @@ web features, such as WebSockets, are not available.
...
+
+For deploying L<Mojolicious> applications on "cloud-native" environments that use L<Envoy|https://www.envoyproxy.io>,
+here is a configuration for a basic reverse proxy similar to Apache and Nginx:
```
Comment (kraih) [inline lib/Mojolicious/Guides/Cookbook.pod:348]: Other sections don't use the "...:" format for examples.

Comment (zakame) [reply lib/Mojolicious/Guides/Cookbook.pod:348]: Yeah, I'll need to rephrase this :muscle:

File: lib/Mojolicious/Guides/Cookbook.pod (hunk 1 above)
Comment (kraih) [reply lib/Mojolicious/Guides/Cookbook.pod]: Looking at the [README](https://github.com/envoyproxy/envoy) it's just referred to as Envoy and not Envoy Proxy. I don't think the "Proxy" is actually part of the name.

**Comment by kraih** [comment]
Instead of making the example too large with more context information, perhaps there's official Envoy docs we could just link to in a second sentence after the example config?

File: lib/Mojolicious/Guides/Cookbook.pod (hunk 1 above)
Comment (zakame) [reply lib/Mojolicious/Guides/Cookbook.pod]: Doesn't matter now, its already dropped :muscle:

**Comment by zakame** [comment]
Yep, added pointers to Istio and Envoy docs for more information.

I can probably make a fuller-fledged example, but that'll take a while.

File: lib/Mojolicious/Guides/Cookbook.pod [hunk 3]
```diff
@@ -342,6 +342,56 @@ web features, such as WebSockets, are not available.
...
+        cluster_name: local_service
+        endpoints:
+        - lb_endpoints:
+          - endpoint:
+              address:
+                socket_address: { address: mojo, port_value: 8080 }
+                
+While this configuration works for simple applications, Envoy's typical use case is for implementing proxies of
+applications as a "service mesh" providing advanced filtering, load balancing, and observability features, such as
+seen in L<Istio|https://istio.io/latest/docs/ops/deployment/architecture/>.  More examples can be found at the
```
Comment (kraih) [inline lib/Mojolicious/Guides/Cookbook.pod:392]: Extra whitespace.

Comment (kraih) [reply lib/Mojolicious/Guides/Cookbook.pod:392]: Is "at the documentation" is correct english?

Comment (zakame) [reply lib/Mojolicious/Guides/Cookbook.pod:392]: Would you prefer
```suggestion
seen in L<Istio|https://istio.io/latest/docs/ops/deployment/architecture/>. For more examples, see the
```

Comment (zakame) [reply lib/Mojolicious/Guides/Cookbook.pod:392]: I'll use the above suggestion then, we can revisit it in the new review pass :muscle:

**Review by marcusramberg — APPROVED** [review]
Lgtm

**Review by Grinnz — APPROVED** [review]
I don't know anything about Envoy, but more cookbook examples are always good.

**Comment by zakame** [comment]
Thanks! :tada:

**Pull request merged**